tests/Agent/IntegrationTests/UnboundedIntegrationTests/Postgres/PostgresIteratorAsyncTests.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using MultiFunctionApplicationHelpers;
+using NewRelic.Agent.IntegrationTestHelpers;
+using NewRelic.Agent.IntegrationTests.Shared;
+using NewRelic.Testing.Assertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NewRelic.Agent.UnboundedIntegrationTests.Postgres
+{
+    public abstract class PostgresSqlIteratorAsyncTestsBase<TFixture> : NewRelicIntegrationTest<TFixture> where TFixture : ConsoleDynamicMethodFixture
+    {
+        private readonly ConsoleDynamicMethodFixture _fixture;
+
+        public PostgresSqlIteratorAsyncTestsBase(TFixture fixture, ITestOutputHelper output) : base(fixture)
+        {
+            _fixture = fixture;
+            _fixture.TestLogger = output;
+
+            _fixture.AddCommand($"PostgresSqlExerciser AsyncIteratorTest");
+
+            _fixture.Actions
+            (
+                setupConfiguration: () =>
+                {
+                    var configPath = fixture.DestinationNewRelicConfigFilePath;
+                    var configModifier = new NewRelicConfigModifier(configPath);
+
+                    configModifier.ForceTransactionTraces()
+                    .SetLogLevel("finest");
+
+                    CommonUtils.ModifyOrCreateXmlAttributeInNewRelicConfig(configPath, new[] { "configuration", "transactionTracer" }, "explainThreshold", "1");
+
+                    var instrumentationFilePath = $@"{fixture.DestinationNewRelicExtensionsDirectoryPath}\NewRelic.Providers.Wrapper.Sql.Instrumentation.xml";
+                    CommonUtils.SetAttributeOnTracerFactoryInNewRelicInstrumentation(instrumentationFilePath, "DataReaderTracer", "enabled", "true");
+                }
+            );
+
+            _fixture.AddActions(exerciseApplication: () => _fixture.AgentLog.WaitForLogLine(AgentLogBase.SqlTraceDataLogLineRegex, TimeSpan.FromMinutes(2)));
+
+            _fixture.Initialize();
+        }
+
+        [Fact]
+        public void Test()
+        {
+            var expectedTransactionName = "OtherTransaction/Custom/MultiFunctionApplicationHelpers.NetStandardLibraries.PostgresSql.PostgresSqlExerciser/AsyncIteratorTest";
+            var expectedDatastoreCallCount = 1;
+
+            //These values are dictated by the queries that are being run as part of this test.
+            //There are two application endpoints being exercised by the test, each of which runs a query that returns a single row.
```

Comment:
Actually, I think the comment is still valid - it's talking about the sequence of calls that occurs during iteration, not that we're calling both the sync and async methods from a single test.

Reply:
The aspect of the comment that talks about how the iterator works is valid, the math in the comment doesn't add up though. Based on the math word problem, the expected call count should be 6, instead of the actual 3.

Reply:
I gotcha now... ;-) Cleaned it up.